File: algorithms/350.py
# https://leetcode.com/problems/intersection-of-two-arrays-ii


class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        i, j, c, o, result = 0, 0, [], [], []
        nums1.sort()
        nums2.sort()
        c, o = nums1, nums2
        while i < len(c) and j < len(o):
            if c[i] < o[j]:
                i += 1
            elif c[i] == o[j]:
                result.append(c[i])
                i += 1
                j += 1
            else:
                j += 1
        return result

File: algorithms/350.swift
// https://leetcode.com/problems/intersection-of-two-arrays-ii

class Solution {
    func intersect(nums1: [Int], _ nums2: [Int]) -> [Int] {
        let n1 = nums1.sort()
        let n2 = nums2.sort()

        var result = [Int]()
        var i = 0
        var j = 0

        while i < n1.count && j < n2.count {
            if n1[i] < n2[j] {
                i += 1
            } else if (n1[i] == n2[j]) {
                result.append(n1[i])
                i += 1
                j += 1
            } else {
                j += 1
            }
        }
        return result
    }
}
